tests/unit/src/test/scala/tests/CodeLensesLspSuite.scala

```diff
@@ -159,6 +159,38 @@ object CodeLensesLspSuite extends BaseLspSuite("codeLenses") {
     } yield ()
   }
 
+  testAsync("refresh-after-error") {
+    for {
+      _ <- server.initialize(
+        """|/metals.json
+           |{ "a": { } }
+           |
+           |/a/src/main/scala/Main.scala
+           |object Main {
+           |  def main(args: Array[String]): Unit = ???
+           |}""".stripMargin
+      )
+      _ <- assertCodeLenses(
+        "a/src/main/scala/Main.scala",
+        """<<run>>
+          |object Main {
+          |  def main(args: Array[String]): Unit = ???
+          |}
+          |""".stripMargin
+      )
+      _ <- server.didSave("a/src/main/scala/Main.scala")(
+        text => text.replace("}", "")
+      )
+      _ <- assertCodeLenses(
+        "a/src/main/scala/Main.scala",
+        """object Main {
```

Comment:
Do I understand correctly that the code lens is removed due to the compile error?

I'm not sure this is desirable behavior. I would expect the code lens to stay stable until a successful compilation invalidates the main/test class.

Reply:
Reasoning: I would like the "run" and "test" code lenses to be as stable as possible to avoid jittery code lenses. If the user requests to run a main or test on broken code then we could do a `window/showMessage` or something like that to explain that it's not possible to run the code because it's not compiling.

Reply:
I can see a value in that and it is consistent with the approach "if it doesn't compile, first - make it compile"